test/parallel/test-fs-open.js

```diff
@@ -32,8 +32,20 @@ fs.open(__filename, 'rs', function(err, fd) {
   openFd2 = fd;
 });
 
+try {
+  fs.open('/path/to/file/that/does/not/exist', { bad: 'flags' }, function() {
+    throw new Error('fs.open callback should not have been called');
+  });
+
+  throw new Error('fs.open did not throw despite bad flags');
+} catch (err) {
+  if (!(err instanceof TypeError)) {
+    throw new Error('unexpected error: ' + err);
+  }
+}
```

Comment:
Understood. If you insist, I'll change it, but I would rather stick to a class check here, to allow other contributions to Node to polish error messages when needed without breaking tests left and right. imho tests shouldn't check for the human readable output when avoidable, and this case it is avoidable.